janitor/timeseries.py

```diff
@@ -87,3 +87,54 @@ def _get_missing_timestamps(
     missing_timestamps = expected_df.index.difference(df.index)
 
     return expected_df.loc[missing_timestamps]
+
+
+@pf.register_dataframe_method
+def range_flag(
+    df: pd.DataFrame,
+    bound: list,
+    inclusive: bool = True
+):
+    """
+    Test each column's values against boundaries.
+    If values in a column,
+    are within bound,
+    this function will return True.
+    If values in a column,
+    are not within bound,
+    this function will return False.
+
+    Example usage:
+    .. code-block:: python
+
+        df = (
+            pd.DataFrame(...)
+            .flag_values_not_in_range(bound=[-3, 5])
+        )
+
+    :param df: dataframe to test for values in range
+    :param bound: left and right boundary
+    :param inclusive: flag to include boundaries
+    :returns: dataframe with booleans for each column
+    """
+    # Check the input data sets for data type
+    check("df", df, [pd.DataFrame])
+    check("bound", bound, [list])
+    check("inclusive", inclusive, [bool])
+
+    # Get names of the columns into a list
+    columns = df.columns
```

Comment:
You can reduce the number of lines and avoid the for loop by using pandas transform, then assigning the result to the new dataframe : 
```python
df1 = df.transform(pd.Series.between, axis=0, left=bound[0], right=bound[-1])
```
You'll have to test the speed though; this however is a built-in pandas function and negates the need for the for loop and the temporary variables for columns.

Also, there is a [between_time](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.between_time.html#:~:text=between_time,-DataFrame.&text=Select%20values%20between%20particular%20times,not%20between%20the%20two%20times.) function as well, with left and right bounds. It could possibly be a better option, since it is dedicated to time manipulations exclusively.

Reply:
@samukweku , 
           I like the idea of keeping it native to Pandas, so I am happy to make this modification, remove the loop and replace it with ```transform()```. The ```between_time``` is only to handle datetimes correct- in my case the data can be anything and not necessarily datetimes. 

@ericmjl,
          Just to make sure I understand your suggestion correctly: 

1. Are you suggesting implementing the function in ```pandera``` package ? 
2. Are you suggesting re-writing the function using ```pandera``` in ```pyjanitor```? 

If it is (2), I believe ```pyjanitor```development environment does not have ```pandera```included. If it is (1), I can get started trying to get familiar with the process of contributing to ```pandera```

Reply:
@UGuntupalli I took a few days to think more carefully about this PR, and I think I was previously getting confused by the PR name. "Testing" whether a column's values are inside a pre-defined range is a different activity from "flagging" whether that column's values are inside a pre-defined range. The difference lies in what actions are taken: a test should error out if the dataframe column fails a check, while a flag will add a column that contains the flag. Tests are out-of-scope for `pyjanitor`, because there's better tooling for this (i.e. `pandera`), but flagging is not, because we already have precedent (`flag_nulls`). Imprecise semantics were what were confusing me before.

The function here is named `flag_range`, which puts it under the flagging category of things. I've renamed the PR and edited the description accordingly. Let's proceed with the PR; you can also ignore my comment about `pandera`, which I typed under the assumption that the intent was to put in a data testing function. (Data tests should be contributed to `pandera` if they aren't available in there already.)